src/client/csfle/state_machine.rs

```diff
@@ -181,6 +209,71 @@ impl CryptExecutor {
     }
 }
 
+#[derive(Debug)]
+struct Mongocryptd {
+    opts: MongocryptdOptions,
+    client: Client,
+    child: Mutex<Result<Process>>,
+}
+
+impl Mongocryptd {
+    async fn new(opts: MongocryptdOptions) -> Result<Self> {
+        let child = Mutex::new(Ok(Self::spawn(&opts)?));
+        let uri = opts.uri.as_deref().unwrap_or("mongodb://localhost:27020");
+        let mut options = crate::options::ClientOptions::parse_uri(uri, None).await?;
+        options.server_selection_timeout = Some(std::time::Duration::from_millis(10_000));
+        let client = Client::with_options(options)?;
+        Ok(Self {
+            opts,
+            client,
+            child,
+        })
+    }
+
+    async fn respawn(&self) -> Result<()> {
```

Comment:
The spec is somewhat quiet about the specifics of this, so I'm going off of DRIVERS-1861 here:
- In the event that `mongocryptd` has already been spawned by another client instance, we'll still want this to return `Ok`. I'm not exactly sure how to detect that though.
- It looks like we'd want to report any other errors related to spawning a process here though.

Reply:
Yeah I don't think we can go that route. I actually just checked it and `spawn` will return `Ok` just fine, so there isn't anything to be done. If we wait for the process to exit and assert its status is how we'd run into issues, but we're not doing that here.